src/Generator/ClassFinder.php

```diff
@@ -0,0 +1,213 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\ApiDocumenter\Generator;
+
+use DateTime as BaseDatetime;
+use DateTimeInterface;
+use EoneoPay\Utils\DateTime;
+use EoneoPay\Utils\UtcDateTime;
+use Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractor;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractorInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+final class ClassFinder
+{
+    /**
+     * @var PropertyInfoExtractorInterface
+     */
+    private $propertyInfo;
+
+    /**
+     * @var string[]
+     */
+    private $skipClasses;
+
+    /**
+     * Constructor.
+     *
+     * @param string[] $skipClasses
```

Comment:
Better docs required on this parameter

Reply:
I think as long as there's an interface that can be bound to in case of a completely different method being used (or caching, or whatever) then I think it's ok.